app/lib/bk/compat/parsers/bitrise.rb

```diff
@@ -41,18 +42,67 @@ def parse
 
         stack_value = @config.dig('meta', 'bitrise.io', 'stack')
         agents = stack_value ? { stack: stack_value } : {}
+
         Pipeline.new(
           steps: bk_steps,
-          agents: agents
+          agents: agents,
+          env: process_app_envs(@config.dig('app', 'envs') || {})
         )
       end
 
+      def process_app_envs(vars)
+        return {} unless vars
+
+        BK::Compat::Bitrise::EnvHelper.process_env_variables(vars)
+      end
+
```

Comment:
`BK::Compat::Bitrise::EnvHelper.process_env_variables` already takes care of empty arrays/hashes and `nil`s so this should be simplified to:

```suggestion
          env: BK::Compat::Bitrise::EnvHelper.process_env_variables(@config.dig('app', 'envs')
      end
```